tests/phpunit/unit/api/actions/beansReplaceActionArguments.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Tests for beans_replace_action_arguments()
+ *
+ * @package Beans\Framework\Tests\Unit\API\Actions
+ *
+ * @since   1.5.0
+ */
+
+namespace Beans\Framework\Tests\Unit\API\Actions;
+
+use Beans\Framework\Tests\Unit\API\Actions\Includes\Replace_Action_Test_Case;
+
+require_once __DIR__ . '/includes/class-replace-action-test-case.php';
+
+/**
+ * Class Tests_BeansReplaceActionArguments
+ *
+ * @package Beans\Framework\Tests\Integration\API\Actions
+ * @group   unit-tests
+ * @group   api
+ */
+class Tests_BeansReplaceActionArguments extends Replace_Action_Test_Case {
+
+	/**
+	 * Test beans_replace_action_arguments() should store the "replaced" action when the original action
+	 * has not yet been registered.
+	 *
+	 * Intent: We are testing to ensure Beans is "load order" agnostic.
+	 */
+	public function test_should_store_when_action_is_not_registered() {
+		$replaced_action = array(
+			'args' => 10,
+		);
+
+		// Store the "replaced" action.
+		foreach ( array_keys( static::$test_actions ) as $beans_id ) {
+			// Test that the original action has not yet been added.
+			$this->assertFalse( _beans_get_action( $beans_id, 'added' ) );
+
+			// Now store away the "replace" args.
+			$this->assertFalse( beans_replace_action_arguments( $beans_id, $replaced_action['args'] ) );
+
+			// Check that it stored.
+			$this->assertEquals( $replaced_action, _beans_get_action( $beans_id, 'modified' ) );
+		}
+	}
+
+	/**
+	 * Test beans_replace_action_arguments() should store the "replaced" hook when the original action
+	 * has not yet been registered.  Once the original action is registered, then the hook should be replaced.
+	 *
+	 * Intent: We are testing to ensure Beans is "load order" agnostic.
+	 */
+	public function test_should_store_and_then_replace_the_args() {
+		$replaced_action = array(
+			'args' => 10,
+		);
+
+		// Now replace the actions.
+		foreach ( static::$test_ids as $beans_id ) {
+			beans_replace_action_arguments( $beans_id, $replaced_action['args'] );
+		}
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $original_action ) {
+			// Check if it replaced the hook.
+			$new_action = _beans_get_action( $beans_id, 'added' );
+			$this->assertEquals( $original_action['hook'], $new_action['hook'] );
+			$this->assertEquals( $original_action['callback'], $new_action['callback'] );
+			$this->assertEquals( $original_action['priority'], $new_action['priority'] );
+			$this->assertEquals( $replaced_action['args'], $new_action['args'] );
+
+			// Check that the "replaced" action has been stored in Beans and WordPress.
+			$this->check_stored_in_beans( $beans_id, $replaced_action );
+			$this->check_registered_in_wp( $original_action['hook'], $new_action );
```

Comment:
Shouldn't this test for the arguments and not the hook?

Reply:
@christophherr Actually, walk through to that method and you'll notice that the 1st parameter is the `'hook` and the 2nd parameter is the action to be tested.  

Here we have replaced the number of arguments but not the hook.

Reply:
@hellofromtonya Of course! Reading code would help... Oh Github interface... What did people do before the onset of IDE's / Code Intellisense?